src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IPatternSwitchCase.cs

```diff
@@ -0,0 +1,679 @@
+    {
+        switch (x)
+        {
+            /*<bind>*/case /*</bind>*/const int y:
+                break;
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+ISwitchStatement (1 cases) (OperationKind.SwitchStatement, IsInvalid) (Syntax: 'switch (x) ... }')
+  Switch expression: IParameterReferenceExpression: x (OperationKind.ParameterReferenceExpression, Type: System.Int32?) (Syntax: 'x')
+  Sections: ISwitchCase (1 case clauses, 2 statements) (OperationKind.SwitchCase, IsInvalid) (Syntax: 'case /*</bi ... break;')
+        Clauses: ISingleValueCaseClause (Equality operator kind: BinaryOperationKind.IntegerEquals) (CaseKind.SingleValue) (OperationKind.SingleValueCaseClause, IsInvalid) (Syntax: 'case /*</bind>*/')
+            IConversionExpression (ConversionKind.Invalid, Implicit) (OperationKind.ConversionExpression, Type: System.Int32, IsInvalid) (Syntax: '')
+              IInvalidExpression (OperationKind.InvalidExpression, Type: ?, IsInvalid) (Syntax: '')
+        Body: IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'const int y')
+            IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'const int y')
+              Variables: Local_1: System.Int32 y
+          IBranchStatement (BranchKind.Break) (OperationKind.BranchStatement) (Syntax: 'break;')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS1525: Invalid expression term 'const'
+                //             /*<bind>*/case /*</bind>*/const int y:
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "const").WithArguments("const").WithLocation(9, 39),
+                // CS1003: Syntax error, ':' expected
+                //             /*<bind>*/case /*</bind>*/const int y:
+                Diagnostic(ErrorCode.ERR_SyntaxError, "const").WithArguments(":", "const").WithLocation(9, 39),
+                // CS0145: A const field requires a value to be provided
+                //             /*<bind>*/case /*</bind>*/const int y:
+                Diagnostic(ErrorCode.ERR_ConstValueRequired, "y").WithLocation(9, 49),
+                // CS1002: ; expected
+                //             /*<bind>*/case /*</bind>*/const int y:
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, ":").WithLocation(9, 50),
+                // CS1513: } expected
+                //             /*<bind>*/case /*</bind>*/const int y:
+                Diagnostic(ErrorCode.ERR_RbraceExpected, ":").WithLocation(9, 50),
+                // CS0168: The variable 'y' is declared but never used
+                //             /*<bind>*/case /*</bind>*/const int y:
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "y").WithArguments("y").WithLocation(9, 49)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<CaseSwitchLabelSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+    }
+}
```

Comment:
Sure, let me add it in a separate PR I am working on currently (https://github.com/dotnet/roslyn/pull/20364)